train_od_ex.py

```diff
@@ -0,0 +1,288 @@
+                np.ones(shape=[self.gt_boxes[idx].shape[0]], dtype=np.int32),
+                dummy_scores, self.category_index)
+        plt.show()
+
+    def create_model_from_checkpoint(self):
+        tf.keras.backend.clear_session()
+
+        print('Building model and restoring weights for fine-tuning...', flush=True)
+        num_classes = 1
+        pipeline_config = 'models/research/object_detection/configs/tf2/ssd_resnet50_v1_fpn_640x640_coco17_tpu-8.config'
+        checkpoint_path = 'models/research/object_detection/test_data/checkpoint/ckpt-0'
+
+        # Load pipeline config and build a detection model.
+        #
+        # Since we are working off of a COCO architecture which predicts 90
+        # class slots by default, we override the `num_classes` field here to be just
+        # one (for our new rubber ducky class).
+        configs = config_util.get_configs_from_pipeline_file(pipeline_config)
+        model_config = configs['model']
+        model_config.ssd.num_classes = num_classes
+        model_config.ssd.freeze_batchnorm = True
+        self.detection_model = model_builder.build(
+            model_config=model_config, is_training=True)
+
+        # Set up object-based checkpoint restore --- RetinaNet has two prediction
+        # `heads` --- one for classification, the other for box regression.  We will
+        # restore the box regression head but initialize the classification head
+        # from scratch (we show the omission below by commenting out the line that
+        # we would add if we wanted to restore both heads)
+        fake_box_predictor = tf.compat.v2.train.Checkpoint(
+            _base_tower_layers_for_heads=self.detection_model._box_predictor._base_tower_layers_for_heads,
+            # _prediction_heads=detection_model._box_predictor._prediction_heads,
+            #    (i.e., the classification head that we *will not* restore)
+            _box_prediction_head=self.detection_model._box_predictor._box_prediction_head,
+        )
+        fake_model = tf.compat.v2.train.Checkpoint(
+            _feature_extractor=self.detection_model._feature_extractor,
+            _box_predictor=fake_box_predictor)
+        ckpt = tf.compat.v2.train.Checkpoint(model=fake_model)
+        ckpt.restore(checkpoint_path).expect_partial()
+
+        # Run model through a dummy image so that variables are created
+        image, shapes = self.detection_model.preprocess(tf.zeros([1, 640, 640, 3]))
+        prediction_dict = self.detection_model.predict(image, shapes)
+        _ = self.detection_model.postprocess(prediction_dict, shapes)
+        print('Weights restored!')
+
+    def finetune(self):
+        tf.keras.backend.set_learning_phase(True)
+
+        # These parameters can be tuned; since our training set has 5 images
+        # it doesn't make sense to have a much larger batch size, though we could
+        # fit more examples in memory if we wanted to.
+        batch_size = 4
```

Comment:
hardcoded numbers. Please, do it in a more explicit way. As a last resort, make global variables.

Reply:
Ok, I'll make an attribute